examples/sampling/graphbolt/rgcn/hetero_rgcn.py

```diff
@@ -580,57 +570,70 @@ def run(
 
         loss = total_loss / num_train
 
-        # Evaluate the model on the train/val/test set.
-        print("Evaluating the model on the training set.")
-        train_acc = evaluate(
-            name, g, model, node_embed, device, train_set, features, num_workers
-        )
-        print("Finish evaluating on training set.")
+        # Evaluate the model on the val/test set.
 
         print("Evaluating the model on the validation set.")
         valid_acc = evaluate(
             name, g, model, node_embed, device, valid_set, features, num_workers
         )
         print("Finish evaluating on validation set.")
 
-        print("Evaluating the model on the test set.")
-        test_acc = evaluate(
-            name,
-            g,
-            model,
-            node_embed,
-            device,
-            test_set,
-            features,
-            num_workers,
-            save_test_submission=(name == "ogb-lsc-mag240m"),
-        )
-        print("Finish evaluating on test set.")
-
-        logger.add_result(run_id, (train_acc, valid_acc, test_acc))
         print(
-            f"Run: {run_id + 1:02d}, "
-            f"Epoch: {epoch +1 :02d}, "
+            f"Epoch: {epoch + 1:02d}, "
             f"Loss: {loss:.4f}, "
-            f"Train: {100 * train_acc:.2f}%, "
-            f"Valid: {100 * valid_acc:.2f}%, "
-            f"Test: {100 * test_acc:.2f}%"
+            f"Valid accuracy: {100 * valid_acc:.2f}%, "
         )
-        print("Finish evaluating on test set.")
 
-    return logger
 
+def infer(
+    name,
+    g,
+    model,
+    node_embed,
+    item_set,
+    device,
+    features,
+    num_workers,
+):
+    model.eval()
+    if name == "ogbn-mag":
+        evaluator = Evaluator(name=name)
+    else:
+        evaluator = MAG240MEvaluator()
+    pred = model.inference(
+        name=name,
+        graph=g,
+        features=features,
+        item_set=item_set,
+        node_embed=node_embed,
+        device=device,
+        num_workers=num_workers,
+    )
+    print(pred)
+    label = item_set._itemsets["paper"]._items[1].to(pred.device)
```

Comment:
access internal data of class object in example is not a good practice.

Reply:
yes, this PR is not ready yet and these unnecessary lines will be removed.